### PR TITLE
switch helm chart URL

### PR DESCRIPTION
**Pull request by juldou**

### CODE REVIEW

File: internal/github/client.go [hunk 1]
```diff
@@ -107,5 +107,5 @@ func (c Client) UploadAsset(ctx context.Context, releaseId int64, release Releas
 
 	opts := &github.UploadOptions{Name: release.AssetPath}
 	asset, _, err := c.gh.Repositories.UploadReleaseAsset(ctx, release.Owner, release.Repo, releaseId, opts, f)
-	return asset.GetBrowserDownloadURL(), err
+	return asset.GetURL(), nil
```
Comment (pete911) [inline internal/github/client.go:110]: pls see the comment in the conversation tab, also we are just returning nil here even if there is error coming from github.

**Comment by juldou** [comment]
We've had issues with the `browser_download_url` - we were unable to download the asset. The name browser suggest it should be used within browser - and yes, that is working correctly - when I paste the link to the browser. With cURL or Helm pull I am getting 404. I found [some](https://gist.github.com/maxim/6e15aa45ba010ab030c4) scripts on the web that can download asset from the private github releases and they're using assets api for that - seems they're also workarounding the issue. When using this assets API, I just need to add `Accept: application/octet-stream` and it downloads the asset. 

You didn't face issues when pulling the chart from helm repo? Anyway, another sad story is that helm doesn't support token authentization - there is [PR](https://github.com/helm/helm/pull/10848) open for long time and seems won't be merged soon.. How do you use this with helm? 

Thank you much for the fast response.

**Comment by pete911** [comment]
@juldou ok, I'll try to test the PR tonight, or tomorrow and will let you know. Is the issue you are facing only on private github repos, or public ones as well?

**Comment by juldou** [comment]
I will need to update this PR soon as we tested end-to-end and found that helm package add filename of tar package to the end of the URL. We haven't tried on public ones, but I would guess that it would work on public ones. Not sure if I will have time to try it on public repo.

**Comment by juldou** [comment]
So helm has problem with URL not ending with filename of tar archive, e.g. `/filename.tgz`. We are developing helm github plugin which will handle this issue, so it will work. The problem is that there is need for that plugin and I'm thinking what is the clear solution here (we need to use assets API).

**Comment by juldou** [comment]
We decided to try going with just that plugin and keep this tool as it is. I will send update later, till then I will convert it to draft so it doesn't have to be reviewed now.

**Comment by pete911** [comment]
cool, keep me updated on the helm plugin pls. I tried to make it work with private repos (index.yaml worked), but I got stuck on helm downloading private releases/assets ... I just realised that the README says it works with private repos, going to fix it.

**Comment by pete911** [comment]
@juldou it would work if the packaged charts are not kept as releases, but they are on the same `gh-pages` branch as `index.yaml`. Then we could use `https://raw.githubusercontent/...` (which supports basic auth) with helm.
Do you see any disadvantage of moving packaged charts from releases to `gh-pages` branch?

**Pull request closed without merging**